src/server.rs

```diff
@@ -1137,12 +1137,12 @@ impl Server {
         result.to_generic::<T::Value>()
     }
 
-    /// Writes value to variable node.
+    /// Writes node value.
     ///
     /// # Errors
     ///
-    /// This fails when the variable node cannot be written.
-    pub fn write_variable(&self, node_id: &ua::NodeId, value: &ua::Variant) -> Result<()> {
+    /// This fails when the node does not exist or its value attribute cannot be written.
+    pub fn write_value(&self, node_id: &ua::NodeId, value: &ua::Variant) -> Result<()> {
```

Comment:
Using a shorter name like `write_value()` instead `write_node_value()` was a deliberate decision? Both client and server might have to deal with other values that do not belong to a node.